lib/Request.js

```diff
@@ -124,101 +133,124 @@ Request.prototype.setQuery = function(query, value) {
+    for (var key in object) {
+        var value = object[key];
 
-    for (var key in query) {
-        // Skip prototype properties, if any
-        if (!query.hasOwnProperty(key)) continue;
+        var filter = new SchemaFilter(this._schema);
+        var schemaProperty = filter.property(key);
 
         // Skip properties that are not in the schema
-        if (!schema.properties.hasOwnProperty(key)) continue;
-
-        if (defaults[key] === '') {
-            // Empty string in defaults means no default, i.e. undefined
-            query[key] = undefined;
+        if (!schemaProperty) {
+            throw new Error("Unexpected query param " + key + " not found in schema.");
         }
 
-        var value = this.getQuery(key);
-        if (typeof value === 'undefined' || value === '') {
+        //If null are allowed and the value is null than don't transform the value
+        if (value == null && schemaProperty.type.indexOf('null') !== -1) {
             continue;
         }
 
-        // We have to do isNaN checks here because NaN can cause
-        // false validation success, because:
-        //
-        // 1) (typeof NaN === 'number') evaluates to `true` and
-        // 2) (NaN < 1) evaluates to `false`
-        switch (schema.properties[key].type) {
-            case 'integer':
-                value = parseInt(value);
-                if (isNaN(value)) return typeError(key);
-                break;
-
-            case 'number':
-                value = parseFloat(value);
-                if (isNaN(value)) return typeError(key);
-                break;
+        if (schemaProperty.type.indexOf('integer') !== -1) {
+            value = parseInt(value);
+
+            if (isNaN(value)) {
+                this._validationErrors.push(typeError(key));
+                return;
+            }
         }
 
-        query[key] = value;
-    }
+        if (schemaProperty.type.indexOf('float') !== -1) {
+            value = parseFloat(value);
+            if (isNaN(value)) {
+                this._validationErrors.push(typeError(key));
+                continue;
+            }
+        }
 
-    var valid = validate(query);
-    this.setQueries(query);
+        if (schemaProperty.type.indexOf('boolean') !== -1) {
+            switch(value) {
+                case 'true' :
+                case true :
+                    value = true;
+                    break
+                case 'false':
+                case false:
+                    value = false
+                    break
+                default:
+                    this._validationErrors.push(typeError(key));
+                    continue;
+            }
+        }
 
-    if (true === valid) {
-        return {
-            valid: true,
-            errors: []
-        };
-    } else {
-        return {
-            valid: false,
-            errors: validate.errors
+        if (schemaProperty.type.indexOf('string') !== -1) {
+            value = value.toString();
```

Comment:
I'm not sure if this will result in the behavior you're expecting:

    > ['a','b'].toString()
    'a,b'
    > {a: 'a', b: 'b'}.toString()
    '[object Object]'
    > null.toString()
    TypeError: Cannot call method 'toString' of null
    > undefined.toString()
    TypeError: Cannot call method 'toString' of undefined